### PR TITLE
Confirm delete attachment

### PR DESCRIPTION
**Pull request by GDSNewt**
When interacting with attachments via the markdown editor the option to "Delete attachment" will now route the user to a confirmation page. Previously clicking this link would have deleted the attachment and displayed a flash message in the modal.  

Screenshot 2020-03-18 at 11 26 08

Adding a confirmation page in this manner brings attachments in line with other delete actions. 

Screenshot 2020-03-18 at 11 25 58

### CODE REVIEW

File: app/assets/javascripts/modules/inline-attachment-modal.js [hunk 1]
```diff
@@ -34,6 +34,9 @@ InlineAttachmentModal.prototype.actionCallback = function (item) {
     'insert': function () {
       this.workflow.render(window.ModalFetch.getLink(item))
     },
+    'confirm-delete': function () {
```
Comment (benthorner) [inline app/assets/javascripts/modules/inline-attachment-modal.js]: I'd actually switch this around with 'delete'. Thinking about it, the 'confirmation' is really on the next page, and the first action is to 'delete'. Does that make sense?

File: app/views/file_attachments/confirm_delete.html.erb [hunk 2]
```diff
@@ -4,7 +4,7 @@
                       file_attachments_path(@edition.document)
                     end %>
 
-<% content_for :back_link, render_back_link(href: back_link_path) %>
+<% content_for :back_link, render_back_link(href: back_link_path, data_attributes: { "modal-action": "back" }) %>
```
Comment (benthorner) [inline app/views/file_attachments/confirm_delete.html.erb]: Good spot ⭐️

File: app/views/file_attachments/confirm_delete.html.erb [hunk 3]
```diff
@@ -25,7 +29,8 @@
       <div class="govuk-body">
         <%= link_to "Cancel",
                   back_link_path,
-                  class: "govuk-link govuk-link--no-visited-state" %>
+                  class: "govuk-link govuk-link--no-visited-state", 
+                  data: { "modal-action": "cancel" } %>
```
Comment (benthorner) [inline app/views/file_attachments/confirm_delete.html.erb]: Minor: I think we expect 'cancel' to behave the same as 'back'. What do you think about re-using 'back' here?

Comment (benthorner) [reply app/views/file_attachments/confirm_delete.html.erb]: Minor: looking at the other four places we have a 'Cancel' action, we should probably use a slightly more specific name here e.g. `cancel-delete-attachment`.

Comment (GDSNewt) [reply app/views/file_attachments/confirm_delete.html.erb]: Using 'back' did occur to me as it is functionally identical but in the end I thought maybe it was good to be a bit more explicit in order to reinforce the idea that they are two seperate user actions. Having said that, I think i'm more inclined just to remove cancel entirely and use back. At least until we develop a policy regarding cancel.

File: app/views/file_attachments/_file_attachment.html.erb [hunk 4]
```diff
@@ -29,19 +29,14 @@
   },
 ) %>
 
-<% actions << capture do %>
-  <%= form_tag(
-    file_attachment_path(document, attachment.file_attachment_id),
-    method: :delete,
-    class: "app-inline-block",
+<% actions << link_to(
+  "Delete attachment",
```
Comment (benthorner) [inline app/views/file_attachments/_file_attachment.html.erb]: Minor: alignment issue here.

File: app/views/file_attachments/confirm_delete.html.erb [hunk 5]
```diff
@@ -30,7 +30,9 @@
         <%= link_to "Cancel",
                   back_link_path,
                   class: "govuk-link govuk-link--no-visited-state", 
-                  data: { "modal-action": "cancel" } %>
+                  data: { "modal-action": "cancel",
+                  gtm: "cancel-delete-attachment"
```
Comment (benthorner) [inline app/views/file_attachments/confirm_delete.html.erb]: Minor: alignment issue here.

**Comment by GDSNewt** [comment]
> Looks good, thanks @GDSNewt! Ideally we should `fixup` those last two "fix" commits into the previous commits, so the commit history is focussed on what we changed. However, I think this PR has been around for long enough, so that's just a suggestion, and I'm happy for this to go in 👏.

Thanks Ben, I've 'fixuped' the two commits. Should be prettier now 😄

**Pull request merged**